### DISCUSSION TITLE
Macros with arguments

### DISCUSSION THREAD

**Discussion by gr-qft** [KaTeX/KaTeX#2589] [General] [answered]
Hi,

I'd really appreciate it if someone could help me with a couple of examples of how to define macros with arguments, when using `renderToString` (see below). I've browsed through the docs (and the file `macros.js`) but I couldn't find anything. Thanks! 

```
 const math = katex.renderToString(expr, {
    displayMode: false,
    macros
  });
```

**Comment by ronkok** [comment, marked as answer]
Try an example:

```
const math = katex.renderToString(expr, {
    displayMode: false,
    macros: {
        "\\addBar": "\\bar{#1}",
        "\\bold": "\\mathbf{#1}"
    }
  });
```

That should enable something like `\addBar{y} + \bold{H}` to return ![MacroExample](https://user-images.githubusercontent.com/16403058/95935282-bbf3df80-0d87-11eb-85af-5f0d1279287f.PNG)

**Comment by gr-qft** [comment]
Oh I see, the keys themselves don't contain arguments, that's why I got complaints from katex. Thank you!
